DP Mystery Map/Assets/Scripts/DialogueManager.cs

```diff
@@ -0,0 +1,110 @@
+using Ink.Parsed;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using Ink.Runtime;
+using PlayerInfo;
+using UnityEngine.UI;
+
+public class DialogueManager : MonoBehaviour
+{
+    [SerializeField] private GameObject dialogPanel;
+    
+    [SerializeField] private TextMeshProUGUI dialogText;
+
+    private Ink.Runtime.Story currentDialog;
+
+    public static bool dialogIsPlaying { get; set; }
+
+    private string fullText;
+    private float charRevealDelay = 0.05f;
+
+    public static DialogueManager instance { get; private set; }
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            Destroy(gameObject); // Ensure only one instance exists
+        }
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+
+        if (Player.collectedItems == Item.None)
+        {
+            dialogIsPlaying = true;
+            dialogPanel.SetActive(true);
+        }
+        else
+        {
+            dialogIsPlaying = false;
+            dialogPanel.SetActive(false);
+        }
+    }
+
+    public void EnterDialogueMode(TextAsset inkJSON)
+    {
+        currentDialog = new Ink.Runtime.Story(inkJSON.text);
+        dialogPanel.SetActive(true);
+        dialogIsPlaying = true;
+
+        ContinueStory();
+    }
+
+    private void ExitDialog() 
+    {
+        dialogText.text = "";
+        dialogPanel.SetActive(false);
+        dialogIsPlaying = false;
+    }
+
+    IEnumerator ShowText()
+    {
+        if (dialogText == null)
+        {
+            Debug.LogError("dialogText is not assigned. Make sure to assign it in the Inspector.");
+            yield break;
+        }
+
+        fullText = currentDialog.Continue(); // Retrieve the next portion of text
+
+        for (int i = 0; i < fullText.Length; i++)
+        {
+            dialogText.text = fullText.Substring(0, i + 1);
+            yield return new WaitForSeconds(charRevealDelay);
+        }
+
+        // The text has been fully revealed
+        yield return null;
+    }
+
+    private void ContinueStory()
+    {
+        if (currentDialog.canContinue)
+        {
+            dialogText.text = currentDialog.Continue();
+            StartCoroutine(ShowText());
+        }
+        else
+            ExitDialog();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (!dialogIsPlaying)
+            return;
+
+
+        if (Input.GetKeyUp(InteractivitySystem.interactKey) && (dialogText.text == fullText))
```

Comment:
Could you change it so that a different key(s) advance the text? Make it a separate static variable in `Player`. Use a list or have 3-4 variables (default keys like space and enter, the left mouse button, and then a player-set key).